lib/fathead/opam/parse.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python
+
+import os
+from urlparse import urljoin
+
+from bs4 import BeautifulSoup
+
+__author__ = "http://github.com/pjhampton"
+__target__ = "https://opam.ocaml.org/"
+__module__ = "OPAM"
+
+class Environment(object):
+  """
+  The Environment class is responsible for setting up the directory
+  and retriving the appropriate files from the web. It is made up of
+  a constructor and a 'setup/0' and 'get_contents/0' member functions.
+  """
+  def __init__(self):
+    print("Environment instanciated")
+    self.setup()
+
+  def setup(self):
+    """
+    The setup member function checks to see if the 'output.txt' file
+    exists. If it does, delete it and move on, otherwise move on. This
+    is called in the constructor of the Environment object and shouldn't
+    be called again.
+    """
+    if os.path.exists("output.txt"):
+      os.remove("output.txt")
+
+  def get_contents(self):
+    """
+    The 'get_contents/0' member function loads the file into memory and
+    reads the contents. The file buffer is then closed and then a
+    BeautifulSoup object is instanciated, with the contents. More on this
+    can be read at the BeautifulSoup documentation website.
+    """
+    f = open('download/packages.html')
```

Comment:
You can leverage a with-statement here to be safe(r) when opening files. Since it is a context manager it'll even close the file after the block is done regardless of exceptions that could be thrown.

```
with open('download/packages.html') as f:
   contents = f.read()
```